R/robustbayesianmetaanalysis.R

```diff
@@ -0,0 +1,2419 @@
+    )
+  } else if (options_prior$type == "gamma_ab") {
+    return(
+      RoBMA::prior(
+        distribution = "gamma",
+        parameters = list(shape      = options_prior$parAlpha,
+                          rate       = options_prior$parBeta),
+        truncation = list(
+          lower      = options_prior$truncationLower,
+          upper      = options_prior$truncationUpper
+        ),
+        prior_odds = options_prior$priorOdds
+      )
+    )
+  } else if (options_prior$type == "gamma_k0") {
+    return(
+      RoBMA::prior(
+        distribution = "gamma",
+        parameters = list(shape      = options_prior$parShape,
+                          scale      = options_prior$parScale2),
+        truncation = list(
+          lower      = options_prior$truncationLower,
+          upper      = options_prior$truncationUpper
+        ),
+        prior_odds = options_prior$priorOdds
+      )
+    )
+  } else if (options_prior$type == "invgamma") {
+    return(
+      RoBMA::prior(
+        distribution = "invgamma",
+        parameters = list(shape      = options_prior$parAlpha,
+                          scale      = options_prior$parBeta),
+        truncation = list(
+          lower      = options_prior$truncationLower,
+          upper      = options_prior$truncationUpper
+        ),
+        prior_odds = options_prior$priorOdds
+      )
+    )
+  } else if (options_prior$type == "spike") {
+    return(
+      RoBMA::prior(
+        distribution = "point",
+        parameters = list(location   = options_prior$parLocation),
+        prior_odds = options_prior$priorOdds
+      )
+    )
+  } else if (options_prior$type == "uniform") {
+    return(
+      RoBMA::prior(
+        distribution = "uniform",
+        parameters = list(a          = options_prior$parA,
+                          b          = options_prior$parB),
+        prior_odds = options_prior$priorOdds
+      )
+    )
+  } else if (options_prior$type %in% c("Two-sided", "Two-sided2")) {
+    return(
+      RoBMA::prior(
+        distribution = "two.sided",
+        parameters = list(alpha      = options_prior$parAlpha,
+                          steps      = options_prior$parCuts),
+        prior_odds = options_prior$priorOdds
+      )
+    )
+  } else if (options_prior$type == "One-sided (mon.)") {
+    return(
+      RoBMA::prior(
+        distribution = "one.sided",
+        parameters = list(alpha      = options_prior$parAlpha,
+                          steps      = options_prior$parCuts),
+        prior_odds = options_prior$priorOdds
+      )
+    )
+  } else if (options_prior$type == "One-sided") {
+    return(
+      RoBMA::prior(
+        distribution = "one.sided",
+        parameters = list(
+          alpha1     = options_prior$parAlpha1,
+          alpha2     = options_prior$parAlpha2,
+          steps      = options_prior$parCuts
+        ),
+        prior_odds = options_prior$priorOdds
+      )
+    )
+  }
+}
+.RoBMA_options2priors_clean <- function(x) {
```

Comment:
What's happening in this function? What's the magic number 666? I have no idea what's going on here. The while loop also looks quite dangerous.

Reply:
Honestly? I have no idea. Probably an easter egg :D 
(removed)

Reply:
I mean the loop with 666.
The rest is just to clean input from the users if they happen to specify the vector of cutoffs / arguments for the Dirichlet distribution in R-like format.

` 
x <- trimws(x, which = "both")  

x <- trimws(x, which = "both", whitespace = "c")  

x <- trimws(x, which = "both", whitespace = "\\(")  

x <- trimws(x, which = "both", whitespace = "\\)")  

x <- trimws(x, which = "both", whitespace = ",")  
`